src/stirling/source_connectors/socket_tracer/metrics.cc

```diff
@@ -28,38 +28,59 @@ namespace px {
 namespace stirling {
 
 SocketTracerMetrics::SocketTracerMetrics(prometheus::Registry* registry,
-                                         traffic_protocol_t protocol)
+                                         traffic_protocol_t protocol, bool tls)
     : data_loss_bytes(
           prometheus::BuildCounter()
               .Name("data_loss_bytes")
               .Help("Total bytes of data loss for this protocol. Measured by bytes that weren't "
                     "successfully parsed.")
               .Register(*registry)
-              .Add({{"protocol", std::string(magic_enum::enum_name(protocol))}})),
+              .Add({
+                  {"protocol", std::string(magic_enum::enum_name(protocol))},
+                  {"tls", tls ? "1" : "0"},
+              })),
       conn_stats_bytes(prometheus::BuildCounter()
                            .Name("conn_stats_bytes")
                            .Help("Total bytes of data tracked by conn stats for this protocol.")
                            .Register(*registry)
-                           .Add({{"protocol", std::string(magic_enum::enum_name(protocol))}})) {}
+                           .Add({
+                               {"protocol", std::string(magic_enum::enum_name(protocol))},
+                               {"tls", tls ? "1" : "0"},
+                           })) {}
 
 namespace {
-std::unordered_map<traffic_protocol_t, std::unique_ptr<SocketTracerMetrics>> g_protocol_metrics;
+std::unordered_map<traffic_protocol_t, std::unique_ptr<SocketTracerMetrics>>
+    g_plaintext_protocol_metrics;
 
-void ResetProtocolMetrics(traffic_protocol_t protocol) {
-  g_protocol_metrics.insert_or_assign(
-      protocol, std::make_unique<SocketTracerMetrics>(&GetMetricsRegistry(), protocol));
+std::unordered_map<traffic_protocol_t, std::unique_ptr<SocketTracerMetrics>>
```

Comment:
Instead of multiple maps can you just create a single map with a compound key

Reply:
I considered this but I was worried about causing memory allocations. I tried to use a `std::pair` as the key, but it seemed like I would need to construct a new object each time I accessed a key since the caller of this code wouldn't have a `std::pair`.

Therefore I chose an implementation that seemed safer. If you think that can be avoided or isn't a concern, I'm definitely open to using a compound key.

Reply:
If you put the `std::pair` on the stack it won't cause any memory allocations. And then whether or not the branch is worse or the couple of instructions for the construction of the pair, seems like a toss up to me, and the pair is probably worth it for keeping the code simpler.

Reply:
Another alternative if you really want to go for performance is to bit pack the enum and the bool into a 64 bit integer key. Since there's no way our enum can have 2**63 values.

Reply:
Would it be reasonable to create a flat array of metrics for each protocol with and w/out ssl? My mental model of a metric suggests that the memory cost of this will be low, and it could lead to simpler reasoning about perf and possibly simpler code also (i.e. removes any of the cruft that comes with handling map corner cases).

Reply:
I incorrectly assumed that `std::make_pair` was going to allocate on the heap when I first looked at this.

> Would it be reasonable to create a flat array of metrics for each protocol with and w/out ssl?
I would prefer to use a complex key since my initial assumption about there being an allocation was incorrect. Unless there are still performance concerns, I think the readability is much nicer.

I needed to provide a specialization in order for the `unordered_map` to understand how to hash a `std::pair`, so I used the enum and bool packing mentioned above as part of this.

Reply:
`std::make_pair` will probably allocate but you can do:
```
std::pair<traffic_protocol_t, bool> key(protocol, ssl);
```
and that will not.

Reply:
I see. I ended up using what you suggested with my latest update, so let me know if there are any concerns with that.